kratos/python_scripts/auxiliar_methods_adaptative_remeshing.py

```diff
@@ -0,0 +1,161 @@
+
+            Keyword arguments:
+            self It signifies an instance of a class.
+        """
+        solver = self.analysis._GetSolver()
+        computing_model_part = solver.GetComputingModelPart()
+        if not self.analysis.process_remesh:
+            convergence_criteria = solver.get_convergence_criterion()
+            convergence_criteria.Initialize(computing_model_part)
+
+        # Ensuring to have conditions on the BC before remesh
+        is_surface = False
+        for elem in computing_model_part.Elements:
+            geom = elem.GetGeometry()
+            if geom.WorkingSpaceDimension() != geom.LocalSpaceDimension():
+                is_surface = True
+            break
+
+        if not is_surface:
+            list_model_parts = []
+            # We need to detect the conditions in the boundary conditions
+            if self.analysis.project_parameters.Has("constraints_process_list"):
+                constraints_process_list = self.analysis.project_parameters["constraints_process_list"]
+                for i in range(0,constraints_process_list.size()):
+                    item = constraints_process_list[i]
+                    list_model_parts.append(item["Parameters"]["model_part_name"].GetString())
+            skin_detection_parameters = KratosMultiphysics.Parameters("""
+            {
+                "list_model_parts_to_assign_conditions" : []
+            }
+            """)
+            root_model_part_name = computing_model_part.GetRootModelPart().Name
+            for name in list_model_parts:
+                name = name.replace(root_model_part_name + ".", "")
+                skin_detection_parameters["list_model_parts_to_assign_conditions"].Append(name)
+
+            if computing_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE] == 2:
+                detect_skin = KratosMultiphysics.SkinDetectionProcess2D(computing_model_part, skin_detection_parameters)
+            else:
+                detect_skin = KratosMultiphysics.SkinDetectionProcess3D(computing_model_part, skin_detection_parameters)
+            detect_skin.Execute()
+        solver.SetEchoLevel(self.analysis.echo_level)
+
+    def AdaptativeRemeshingRunSolutionLoop(self):
+        """This function executes the solution loop of the AnalysisStage for cases where remeshing may be considered
+
+            Keyword arguments:
+            self It signifies an instance of a class.
+        """
+
+        # If we remesh using a process
+        computing_model_part = self.analysis._GetSolver().GetComputingModelPart()
+        root_model_part = computing_model_part.GetRootModelPart()
+
+        while self.analysis.KeepAdvancingSolutionLoop():
+            self.analysis.time = self.analysis._GetSolver().AdvanceInTime(self.analysis.time)
+            # We reinitialize if remeshed previously
+            if root_model_part.Is(KratosMultiphysics.MODIFIED):
+                self.analysis.ReInitializeSolver()
```

Comment:
Can you not make this call inside a reimplementation of ExecuteInitializeSolutionStep? in a remeshing analysis that derives from the base one?

Reply:
OK, maybe you are right